### PR TITLE
Throw exception for non-absolute paths in image factory

### PR DESCRIPTION
**Pull request by ausi**
Calling `$imageFactory->create()` with a relative path can lead to ambiguous error messages because non-existent files are now possible with deferred images.

I think we should catch this error early to generate a meaningful error exception.

### CODE REVIEW

**Review by Toflar — APPROVED** [review]

**Review by aschempp — APPROVED** [review]
The interface description always said the path must be absolute, so that should be fine. Should we do the same in the picture factory?

**Comment by ausi** [comment]
> Should we do the same in the picture factory?

Isn’t necessary I think because the picture factory directly calls `$this->imageFactory->create($path)`.

File: core-bundle/tests/Image/ImageFactoryTest.php [hunk 1]
```diff
@@ -449,7 +458,7 @@ public function testCreatesAnImageObjectFromAnImagePathInLegacyMode(string $mode
         $path = $this->getFixturesDir().'/images/none.jpg';
         $imageMock = $this->createMock(ImageInterface::class);
 
-        $filesystem = $this->createMock(Filesystem::class);
+        $filesystem = $this->getMockBuilder(Filesystem::class)->setMethods(['exists'])->getMock();
```
Comment (leofeyer) [inline core-bundle/tests/Image/ImageFactoryTest.php]: Why is this necessary?

Comment (ausi) [reply core-bundle/tests/Image/ImageFactoryTest.php]: Because I don’t want to mock `isAbsolutePath()` because it would returns `null` otherwise.

**Comment by leofeyer** [comment]
Thank you @ausi.
